index/Dockerfile

```diff
@@ -14,21 +20,23 @@ RUN apt-get update \
         unzip \
         # Build tools\
         build-essential \
-        python3-pip \
         python3-dev \
         # For Psycopg2
-        libpq-dev\
+        libpq-dev \
         # Yaml parsing speedup, I think
         libyaml-dev \
         lsb-release \
         # for shapely with --no-binary
         libgeos-dev \
-        postgresql-client-14 \
+        postgresql-client-16 \
     # Cleanup
     && apt-get autoclean \
     && apt-get autoremove \
     && rm -rf /var/lib/{apt,dpkg,cache,log}
 
+ENV VIRTUAL_ENV /virtualenv/python3.12
+ENV PATH /virtualenv/python3.12/bin:$PATH
+
```

Comment:
Do we need to set these manually? Isn't sourcing the activate script sufficient?  Or does the way Docker uses intermediate containers break that approach?

Reply:
I guess if it's passing tests it's probably OK.